### DISCUSSION TITLE
PR replacing joi dependency with @hapi/joi which is depreacted

### DISCUSSION THREAD

**Discussion by dkaksl** [renovatebot/renovate#31148] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

bitbucket cloud (38.53.0)

### Please tell us more about your question or problem

I got a PR with the subject `[renovate] Replace dependency joi with @hapi/joi ^14.4.0`, replacing `joi@^14.3.1` with `@hapi/joi@^14.4.0`. This doesn't seem to work as a drop-in replacement, and @hapi/joi is deprecated in npm.

![image](https://github.com/user-attachments/assets/3fa2b211-8783-44cc-a75b-d63a0d3c508c)

Is this a bug? If not, what is the logic behind this dependency switch?

**Comment by rarkins** [comment]
It looks like the package maybe got moved around? https://github.com/renovatebot/renovate/blame/058578f619f085781f0e0860a5e38e09801be838/lib/data/replacements.json

**Reply by rarkins** [reply]
Looks like `joi` got moved to `@hapi/joi` from 14.4.0 until v17. There was a big gap in `joi` versions: 
<img width="790" alt="image" src="https://github.com/user-attachments/assets/f857bc20-8d14-4aa7-a53e-dce19fc04e33">
